File: benchmark/c/cpw/sum_pairwise_products_1/sum_pairwise_products_1.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int sum_pairwise_products_1 ( int n ) ;
int sum_pairwise_products_1 ( int n ) {
  long int multiTerms = n * ( n + 1 ) / 2;
  long int sum = multiTerms;
  for ( int i = 2;
  i <= n;
  i ++ ) {
    multiTerms = multiTerms - ( i - 1 );
    sum = sum + multiTerms * i;
  }
  return sum;
}

File: translation_result/else_bms/fail/c/cpw/sum_pairwise_products_1/gemini/CAPR/sum_pairwise_products_1.rs

fn sum_pairwise_products_1(n: i32) -> i32 {
    let mut multi_terms = (n * (n + 1)) / 2;
    let mut sum = multi_terms;
    for i in 2..=n {
        multi_terms -= i - 1;
        sum += multi_terms * i;
    }
    sum
}
